File: T4/codigos2122/Dialogos/app/src/main/java/com/example/dialogos/dialogos/DialogoHora.java
package com.example.dialogos.dialogos;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import java.util.Calendar;

public class DialogoHora extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        return new TimePickerDialog(getContext(), (TimePickerDialog.OnTimeSetListener) getContext(), Calendar.getInstance().get(Calendar.HOUR_OF_DAY),Calendar.getInstance().get(Calendar.MINUTE),true);
    }
}

File: Kotlin/T4/codigos_ret/T4_ComunicacionDialogo/app/src/main/java/com/example/t4_comunicaciondialogo/ui/dialogs/DialogoHora.kt
package com.example.t4_comunicaciondialogo.ui.dialogs

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.content.Context
import android.os.Bundle
import android.provider.CalendarContract.CalendarAlerts
import androidx.fragment.app.DialogFragment
import java.util.Calendar

class DialogoHora : DialogFragment() {

    private lateinit var contexto: Context

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.contexto = context;
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val calendario = Calendar.getInstance();
        return TimePickerDialog(
            contexto, contexto as OnTimeSetListener,
            calendario.get(Calendar.HOUR_OF_DAY),
            calendario.get(Calendar.MINUTE), true
        )
    }

}